syntax = "proto3";

// ============================================================================
// Enums
// ============================================================================

enum Trip {
  UNKNOWN_TRIP = 0;
  ROUND_TRIP = 1;
  ONE_WAY = 2;
  MULTI_CITY = 3;
}

enum Seat {
  UNKNOWN_SEAT = 0;
  ECONOMY = 1;
  PREMIUM_ECONOMY = 2;
  BUSINESS = 3;
  FIRST = 4;
}

enum Passenger {
  UNKNOWN_PASSENGER = 0;
  ADULT = 1;
  CHILD = 2;
  INFANT_IN_SEAT = 3;
  INFANT_ON_LAP = 4;
}

// ============================================================================
// Messages
// ============================================================================

// Airport identifier (e.g., "SFO", "JFK")
// Cities decode to deterministic strings like "/m/07dfk"
message Airport {
  string airport = 2;
}

// Individual flight segment information
message FlightData {
  string date = 2;              // Flight date in YYYY-MM-DD format
  optional int32 max_stops = 5; // Maximum number of stops (null = any)
  repeated string airlines = 6; // List of airline codes
  Airport from_flight = 13;     // Departure airport
  Airport to_flight = 14;       // Arrival airport
}

// Complete flight search information
message Info {
  repeated FlightData data = 3;     // Flight segments
  repeated Passenger passengers = 8; // List of passengers
  Seat seat = 9;                     // Seat class preference
  Trip trip = 19;                    // Type of trip
}